li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/common/LargeMessageHeaderValue.java

```diff
@@ -110,7 +108,7 @@ public static LargeMessageHeaderValue fromBytes(byte[] bytes) {
     int segmentNumber = PrimitiveEncoderDecoder.decodeInt(bytes, byteOffset);
     byteOffset += PrimitiveEncoderDecoder.INT_SIZE;
     int numberOfSegments = PrimitiveEncoderDecoder.decodeInt(bytes, byteOffset);
-    if (bytes.length == LEGACY_V2_HEADER_SIZE) {
+    if (bytes.length == LEGACY_V3_HEADER_SIZE) {
```

Comment:
use type here?
if (type > LEGACY)?
alternatively:
if (bytes.length == LEGACY_V2_HEADER_SIZE || bytes.length == LEGACY_V3_HEADER_SIZE) 
==> so that we know it covers both cases.

Reply:
updated